internal/processing/processing.go

```diff
@@ -59,15 +60,26 @@ func (f *Factory) CalculateRequiredState(api *gatewayv1beta1.APIRule, config *he
 	var res State
 	pathDuplicates := hasPathDuplicates(api.Spec.Rules)
 	res.accessRules = make(map[string]*rulev1alpha1.Rule)
+	res.authorizationPolicies = make(map[string]*istiosecurityv1beta1.AuthorizationPolicy)
+	res.requestAuthentications = make(map[string]*istiosecurityv1beta1.RequestAuthentication)
+
 	for _, rule := range api.Spec.Rules {
 		if isSecured(rule) {
 			var ar *rulev1alpha1.Rule
+			var ap *istiosecurityv1beta1.AuthorizationPolicy
+			var ra *istiosecurityv1beta1.RequestAuthentication
+
 			if config.JWTHandler == helpers.JWT_HANDLER_ORY {
 				ar = generateAccessRule(api, rule, rule.AccessStrategies, f.additionalLabels, f.defaultDomainName)
 			} else if config.JWTHandler == helpers.JWT_HANDLER_ISTIO {
 				//TODO generate based on config.JWTHandler="istio"
+				ap = generateAuthorizationPolicy(api, rule, f.additionalLabels)
+				ra = generateRequestAuthentication(api, rule, f.additionalLabels)
 			}
+
```

Comment:
should test here for nil dereference on `ar`, `ap` and `ra`, before further process on them